### PR TITLE
Add OStream metrics exporter

### PR DESCRIPTION
**Pull request by HudsonHumphries**
This PR implements the OStream Exporter for the metrics data pipeline. This MetricsExporter takes data in from the Controller in the form of Records. The exporter then takes the records and gets all of the relevant data out of it and sends it to an ostream. The default configuration is to send the recordable data to stdout. Unlike SpanExporters, metrics exporters are currently not required to implement a Shutdown() function.

**This PR contains a lot of code from PRs currently open. These should either be removed from this PR before merging or should be merged before this one is. Please only review the exporter portion of this PR and redirect all feedback on Aggregators to their specific PRs.**

CI Tests were not running on the old PR, so I made a new one.

### CODE REVIEW

**Comment by HudsonHumphries** [comment]
@reyang @pyohannes This PR should be ready for review and can be merged once the Sketch Aggregator is merged.

File: sdk/include/opentelemetry/sdk/metrics/aggregator/aggregator.h [hunk 1]
```diff
@@ -45,7 +45,6 @@ class Aggregator
    * @return none
    */
   virtual void update(T val) = 0;
-
```
Comment (reyang) [inline sdk/include/opentelemetry/sdk/metrics/aggregator/aggregator.h:48]: Are these intentional?

File: exporters/ostream/include/opentelemetry/exporters/ostream/metrics_exporter.h [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+   * it is holding. Each Aggregator holds data differently, so each have their own
+   * custom printing.
+   */
+  template <typename T>
+  void PrintAggregatorVariant(sdkmetrics::AggregatorVariant value)
+  {
+    auto agg     = nostd::get<std::shared_ptr<sdkmetrics::Aggregator<T>>>(value);
+    auto aggKind = agg->get_aggregator_kind();
+
+    if (agg != nullptr)
```
Comment (reyang) [inline exporters/ostream/include/opentelemetry/exporters/ostream/metrics_exporter.h:47]: Could we simply the flow here:
1. short circuit if `agg == nullptr`.
2. use `switch case` for the conditions.

So the flow would look like:
```c
if (!agg) return;
switch (aggKind) {
case x:
  // do something
  break;
}
```

File: sdk/include/opentelemetry/sdk/metrics/aggregator/aggregator.h (hunk 1 above)
Comment (HudsonHumphries) [reply sdk/include/opentelemetry/sdk/metrics/aggregator/aggregator.h:48]: This happened when running the format tool, it's not intentional. I can remove

**Comment by HudsonHumphries** [comment]
Addressed comments and rebased @reyang Let me know if I didn't address your comments.

**Comment by HudsonHumphries** [comment]
Design document can be found [here](https://github.com/open-o11y/docs/blob/master/exporter/cpp-ostream/ostream-exporter-design.md)

File: sdk/include/opentelemetry/sdk/metrics/exporter.h [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+   * Exporting failed. The caller must not retry exporting the same batch; the
+   * batch must be dropped.
+   */
+  kFailure
+};
+/**
+ * MetricsExporter defines the interface that protocol-specific span exporters must
+ * implement.
+ */
+class MetricsExporter
```
Comment (reagankan) [inline sdk/include/opentelemetry/sdk/metrics/exporter.h:31]: unc really be manifesting MC and scrapers since the start of the decade

**Pull request merged**